Make audit webhook and kafka config dynamic

anjalshireesh opened the pull request:
## Description

Make the audit webhook and kafka config dynamic

## Motivation and Context

These are similar to the logger webhook config which is dynamic,
and there is no reason for them to not be.

## How to test this PR?

- Test that changing value of an audit webhook or audit kafka config using `mc admin config set {alias} audit_webhook|audit_kafka` doesn't ask you to restart the server
- Test that `get, set and reset` of all configs is working fine

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

anjalshireesh committed:
Make audit webhook config dynamic

anjalshireesh committed:
Make audit kafka config also dynamic

anjalshireesh committed:
rename targets to httpTargets

harshavardhana commented on internal/logger/targets.go:
```diff
@@ -131,15 +116,28 @@ func initTargets(cfg Config) (tgts []Target, err error) {
 
 	swapMu.Lock()
-	for _, tgt := range targets {
+	for _, tgt := range httpTargets {
```
Looks like regression is coming in again @anjalshireesh can you test these things locally? so that we can avoid further regressions on console logging?

harshavardhana reviewed (commented):
PR looks good other than some variables that can be renamed for brevity other than that its works fine.

anjalshireesh committed:
Rename httpTargets to systemTargets

klauspost commented on internal/logger/targets.go:
```diff
@@ -148,8 +146,76 @@ func UpdateTargets(cfg Config) error {
+	for _, tgt := range auditTargets {
+		if tgt.Endpoint() != kafka.NAME {
+			tgt.Cancel()
+		}
```
This is a bit hacky, and a trap for the future.

Add a `Type() TargetType` that allow you to get the exact underlying type, and don't duplicate code:

```
type TargetType uint8

func existingTargets(t TargetType) []Target {
	awTgts := make([]Target{}, 0, len(auditTargets))
	for _, tgt := range auditTargets {
		if tgt.Type() == t {
			awTgts = append(awTgts, tgt)
		}
	}
	return awTgts
}

func cancelTargetType(t TargetType) {
	for _, tgt := range auditTargets {
		if tgt.Type() ==t {
			tgt.Cancel()
		}
	}
}

```

anjalshireesh replied:
Done.

anjalshireesh committed:
Introduce a new type called TargetType

And use it to get existing audit targets of a given type, rather than
comparing the endpoint with "kafka".
